### PR TITLE
Verify API resources and log totals

### LINKED ISSUE #46: Generate a manifest to check builds
During each build, generate a manifest so that we can double-check the number of generated pages (of different types) against the original database.

### PR DESCRIPTION
**Pull request by eatyourgreens**
- Add the `--quiet` flag to production builds and remove some of the noisier logging.
- Verify users and subjects against the URL that was used to request them.
- Log total numbers of resources loaded for discussions, boards, collections, subjects and users.
- Publish a manifest at `/manifest/build.json`, with expected total counts for each resource type.
- Verify the number of output files after each build.

Closes #46.

### CODE REVIEW

**Comment by eatyourgreens** [comment]
Build output now looks something like this. It's much quieter, but there's still a lot of repetition from the parallel builds.
```
$ NODE_ENV=production node --max-old-space-size=${NODE_RAM:-12000} ./node_modules/@11ty/eleventy/cmd.js --input=src/boards --quiet talk.notesfromnature.org
$ NODE_ENV=production node --max-old-space-size=${NODE_RAM:-12000} ./node_modules/@11ty/eleventy/cmd.js --input=src/subjects --quiet talk.notesfromnature.org
$ NODE_ENV=production node --max-old-space-size=${NODE_RAM:-12000} ./node_modules/@11ty/eleventy/cmd.js --input=src/api --quiet talk.notesfromnature.org
$ NODE_ENV=production node --max-old-space-size=${NODE_RAM:-12000} ./node_modules/@11ty/eleventy/cmd.js --quiet talk.notesfromnature.org
$ NODE_ENV=production node --max-old-space-size=${NODE_RAM:-12000} ./node_modules/@11ty/eleventy/cmd.js --input=src/tags --quiet talk.notesfromnature.org
read 970 users
read 970 users
read 10865 subjects
read 10865 subjects
loaded 5 boards
loaded 875 discussions
read 970 users
Bad response for https://api.zooniverse.org/projects/notes_from_nature/talk/users/52ab42fee3833a4d3400005f (404): Not Found
Bad response for https://api.zooniverse.org/projects/notes_from_nature/talk/users/5535614e8362d665e00005b3 (404): Not Found
Bad response for https://api.zooniverse.org/projects/notes_from_nature/talk/users/52ab42fee3833a4d3400005f (404): Not Found
Bad response for https://api.zooniverse.org/projects/notes_from_nature/talk/users/5535614e8362d665e00005b3 (404): Not Found
Bad response for https://api.zooniverse.org/projects/notes_from_nature/talk/users/52ab42fee3833a4d3400005f (404): Not Found
Bad response for https://api.zooniverse.org/projects/notes_from_nature/talk/users/5535614e8362d665e00005b3 (404): Not Found
loaded 968 users
read 152 collections
loaded 968 users
read 152 collections
loaded 968 users
Benchmark (Data): `./src/api/templates/collection.11tydata.js` took 6900ms (37.6%, called 153×, 45.1ms each)
Benchmark (Data): `./src/api/templates/user.11tydata.js` took 6849ms (37.3%, called 969×, 7.1ms each)
Copied 1 file / Wrote 1121 files in 17.94 seconds (16.0ms each, v0.11.0)
loaded 10865 subjects
read 970 users
loaded 10865 subjects
read 970 users
Bad response for https://api.zooniverse.org/projects/notes_from_nature/talk/users/52ab42fee3833a4d3400005f (404): Not Found
Bad response for https://api.zooniverse.org/projects/notes_from_nature/talk/users/5535614e8362d665e00005b3 (404): Not Found
Bad response for https://api.zooniverse.org/projects/notes_from_nature/talk/users/52ab42fee3833a4d3400005f (404): Not Found
Bad response for https://api.zooniverse.org/projects/notes_from_nature/talk/users/5535614e8362d665e00005b3 (404): Not Found
loaded 968 users
loaded 968 users
read 152 collections
loaded 5 boards
loaded 875 discussions
Benchmark (Data): `./src/boards/_data/users.js` took 6450ms (12.7%)
Copied 1 file / Wrote 1762 files in 50.36 seconds (28.6ms each, v0.11.0)
read 416 tags
Benchmark (Data): `./src/tags/_data/subjects.js` took 35091ms (50.0%)
Benchmark (Data): `./src/tags/_data/userCollections.js` took 7394ms (10.5%)
Benchmark (Data): `./src/tags/_data/userTags.js` took 11811ms (16.8%)
Copied 1 file / Wrote 2080 files in 69.82 seconds (33.6ms each, v0.11.0)
Copied 1 file / Wrote 3057 files in 124.92 seconds (40.9ms each, v0.11.0)
Benchmark (Data): `./src/subjects/_data/subjects.js` took 34091ms (8.3%)
Copied 1 file / Wrote 21730 files in 409.94 seconds (18.9ms each, v0.11.0)
✨  Done in 419.51s.
```

**Comment by eatyourgreens** [comment]
I've added a script that verifies the number of `.json` files in `/dist/api`. It could maybe count `.html` files instead, in case the HTML files don't build but the JSON files do.

Screenshot of the build log, showing the final log step which checks the number of output files.

File: verify.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+boards.forEach(board => {
+  const discussionsPath = path.resolve(source, board.name, 'discussions');
+  const discussions = fs.readdirSync(discussionsPath, { withFileTypes: true }).filter(dirent => dirent.isDirectory());
+  discussionsCount = discussionsCount + discussions.length;
+});
+
+if (discussionsCount === manifest.discussions) {
+  console.log(logSymbols.success, 'discussions', discussionsCount);
+} else {
+  console.error(logSymbols.error, 'discussions', discussionsCount, '(', manifest.discussions, ')');
```
Comment (eatyourgreens) [inline verify.js:51]: Should we raise an error if verification fails?

**Comment by camallen** [comment]
Nice - this is looking good. Could we add in the expected counts e.g. `boards 11 / 11` ?

This is what i've got from the raw export file counts, it seems subject discussions merge to the subject pages?
```
$ wc -l illustratedlife_boards.json 
     11 illustratedlife_boards.json

$ wc -l illustratedlife_collections.json 
     627 illustratedlife_collections.json

$ wc -l illustratedlife_discussions.json 
   10687 illustratedlife_discussions.json

$ wc -l illustratedlife_subject_ids.csv 
   11871 illustratedlife_subject_ids.csv

$ wc -l illustratedlife_user_ids.csv 
     987 illustratedlife_user_ids.csv
```

**Comment by eatyourgreens** [comment]
Yeah, `subject.discussion` is the list of comments attached to a subject eg. the two comments here: https://talk.sciencegossip.org/#/subjects/ASC0002rfl

I don't think those discussions get referenced by ID anywhere, because they don't have boards.

Collections also have discussions, which are listed as comments with the collection and likewise don't have boards.
https://talk.penguinwatch.org/#/collections/CPZS0002zu

**Pull request merged**